### PR TITLE
Fixes #30532 - CV publish with separate package env copy task

### PR DESCRIPTION
**Pull request by sjha4**
Currently blocked on https://pulp.plan.io/issues/7248

To test:
1. Set up pulp_rpm 3.5.0 and then apply this patch https://github.com/pulp/pulp_rpm/pull/1794 and restart pulp
2. Sync RHEL 8 Appstream repo
3. Create content view with any filter and publish.
4. Look at the pulp3 data with something like:
curl --cert /etc/pki/katello/certs/pulp-client.crt --key /etc/pki/katello/private/pulp-client.key https://`hostname`/pulp/api/v3/repositories/rpm/rpm/`997aa028-3467-47a5-95b3-d92dbaf43b1e`/versions/ | python -m json.tool
of the target repo to make sure all package environment were added to the latest version.

### CODE REVIEW

File: app/services/katello/pulp3/repository/yum.rb [hunk 1]
```diff
@@ -104,14 +100,32 @@ def copy_units(source_repository, content_unit_hrefs, dependency_solving, dest_b
...
           else
             tasks << remove_all_content
           end
-
           tasks
         end
 
+        def copy_package_env(source_repository, dest_base_version)
+          options = { :repository_version => source_repository.version_href }
+          package_env_hrefs = packageenvironments(options).results.map(&:pulp_href)
```
Comment (jlsherrill) [inline app/services/katello/pulp3/repository/yum.rb:114]: could there be more than a page_size of results?  Probably want to use this:  https://github.com/Katello/katello/blob/master/app/services/katello/pulp3/api/core.rb#L145

Comment (jlsherrill) [reply app/services/katello/pulp3/repository/yum.rb:114]: oh and you might wanna sort href list that avoid testing issues :)

**Comment by jjeffers** [comment]
Applied the suggested patch to the pulp_rpm 3.5.0 installation. Then created the repository as mentioned, synced it, published a content view with that repository.

```[vagrant@centos7-katello-devel-stable ~]$ curl --cert /etc/pki/katello/certs/pulp-client.crt --key /etc/pki/katello/private/pulp-client.key https://`hostname`/pulp/api/v3/repositories/rpm/rpm/168e27a8-3574-4112-acef-a1b2f0f66e28/versions/1/ | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3239  100  3239    0     0   9510      0 --:--:-- --:--:-- --:--:--  9526
{
    "base_version": null,
    "content_summary": {
        "added": {
            "rpm.advisory": {
                "count": 759,
                "href": "/pulp/api/v3/content/rpm/advisories/?repository_version_added=/pulp/api/v3/repositories/rpm/rpm/168e27a8-3574-4112-acef-a1b2f0f66e28/versions/1/"
            },
            [...]
            "rpm.packageenvironment": {
                "count": 4,
                "href": "/pulp/api/v3/content/rpm/packageenvironments/?repository_version=/pulp/api/v3/repositories/rpm/rpm/168e27a8-3574-4112-acef-a1b2f0f66e28/versions/1/"
            },
            "rpm.packagegroup": {
                "count": 58,
                "href": "/pulp/api/v3/content/rpm/packagegroups/?repository_version=/pulp/api/v3/repositories/rpm/rpm/168e27a8-3574-4112-acef-a1b2f0f66e28/versions/1/"
            },
            "rpm.repo_metadata_file": {
                "count": 1,
                "href": "/pulp/api/v3/content/rpm/repo_metadata_files/?repository_version=/pulp/api/v3/repositories/rpm/rpm/168e27a8-3574-4112-acef-a1b2f0f66e28/versions/1/"
            }
        },
        "removed": {}
    },
    "number": 1,
    "pulp_created": "2020-08-04T20:40:09.967071Z",
    "pulp_href": "/pulp/api/v3/repositories/rpm/rpm/168e27a8-3574-4112-acef-a1b2f0f66e28/versions/1/"
```

It appears that the package environment groups were copied into the latest version.

**Comment by sjha4** [comment]
Cherry-picking the optimize PR commit 9d667531ec35487f02d8ad4256782662b95af070 on this so we can merge everything together..

**Comment by sjha4** [comment]
Merging without squashing to keep commits separate.

**Pull request merged**